h2o-algos/src/test/java/hex/adaboost/AdaBoostTest.java

```diff
@@ -0,0 +1,489 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.EnvironmentVariables;
+import org.junit.runner.RunWith;
+import water.DKV;
+import water.Scope;
+import water.TestUtil;
+import water.fvec.Frame;
+import water.fvec.TestFrameBuilder;
+import water.fvec.Vec;
+import water.runner.CloudSize;
+import water.runner.H2ORunner;
+import water.util.FrameUtils;
+
+import java.io.File;
+import java.util.Arrays;
+
+import static org.junit.Assert.*;
+
+@CloudSize(1)
+@RunWith(H2ORunner.class)
+public class AdaBoostTest extends TestUtil {
+
+    @Rule
+    public EnvironmentVariables environmentVariables = new EnvironmentVariables();
+
+    @Before
+    public void beforeClass() {
+        final File h2oHomeDir = new File(System.getProperty("user.dir")).getParentFile();
+        environmentVariables.set("H2O_FILES_SEARCH_PATH", h2oHomeDir.getAbsolutePath());
+    }
+    
+    @Test
+    public void testBasicTrain() {
+        try {
+            Scope.enter();
+            Frame train = parseTestFile("smalldata/prostate/prostate.csv");
+            String response = "CAPSULE";
+            train.toCategoricalCol(response);
+            Scope.track(train);
+            AdaBoostModel.AdaBoostParameters p = new AdaBoostModel.AdaBoostParameters();
+            p._train = train._key;
+            p._seed = 0xDECAF;
+            p._nlearners = 50;
+            p._response_column = response;
+
+            AdaBoost adaBoost = new AdaBoost(p);
+            AdaBoostModel adaBoostModel = adaBoost.trainModel().get();
+            Scope.track_generic(adaBoostModel);
+            assertNotNull(adaBoostModel);
+
+            for (int i = 0; i < adaBoostModel._output.models.length; i++) {
+                System.out.println("Tree = " + i);
+                DRFModel drfModel = DKV.getGet(adaBoostModel._output.models[i]);
+                SharedTreeSubgraph tree = drfModel.getSharedTreeSubgraph(0,0);
+                if (tree.rootNode.getColName() == null) {
+                    // FIXME - why are some of the trees empty? Are all of the columns bad for split?
+                    System.out.println("    Empty tree");
+                    continue;
+                }
+                System.out.println("    Root = " + tree.rootNode.getColName() + " " + tree.rootNode.getSplitValue());
+                System.out.println("    Left = " + tree.rootNode.getLeftChild().isLeaf() + " " + tree.rootNode.getLeftChild().getPredValue());
+                System.out.println("    Right = " + tree.rootNode.getRightChild().isLeaf() + " " + tree.rootNode.getRightChild().getPredValue());
+                assertNotNull(tree.rootNode.getColName());
+                assertTrue(tree.rootNode.getLeftChild().isLeaf());
+                assertTrue(tree.rootNode.getRightChild().isLeaf());
+            }
+        } finally {
+            Scope.exit();
+        }
+    }
+
+    @Test
+    public void testBasicTrainGLM() {
+        try {
+            Scope.enter();
```

Comment:
I would expect this test to fail because of: https://github.com/h2oai/h2o-3/pull/15639/files#r1333015134